migrations/versions/20181212_rename_page_table.py

```diff
@@ -0,0 +1,160 @@
+    op.drop_constraint("uq_page_guid_version", "page", type_="unique")
+    op.drop_constraint("page_guid_version_pkey", "page", type_="primary")
+    op.drop_constraint("data_source_id_page_guid_version_pk", "data_source_in_page", type_="primary")
+
+    # * create `measure_version_id_seq`
+    op.execute(CreateSequence(Sequence("measure_version_id_seq")))
+
+    # * add columns
+    #   * `id` to `page`, not nullable, default next from `measure_version_id_seq`
+    op.add_column(
+        "page",
+        sa.Column(
+            "id", sa.Integer(), nullable=False, server_default=sa.text("nextval('measure_version_id_seq'::regclass)")
+        ),
+    )
+    op.add_column("page", sa.Column("parent_id", sa.Integer(), nullable=True))
+    op.add_column("data_source_in_page", sa.Column("measure_version_id", sa.Integer(), nullable=True))
+    op.add_column("upload", sa.Column("measure_version_id", sa.Integer(), nullable=True))
+    op.add_column("dimension", sa.Column("measure_version_id", sa.Integer(), nullable=True))
+
+    conn = op.get_bind()
+
+    conn.execute(
+        "UPDATE page AS child SET parent_id = parent.id FROM page AS parent WHERE child.parent_guid = parent.guid AND child.parent_version = parent.version AND child.page_type = 'measure' and parent.page_type = 'measure'"
+    )
+    conn.execute(
+        "UPDATE upload SET measure_version_id = page.id FROM page WHERE upload.page_id = page.guid AND upload.page_version = page.version"
+    )
+    conn.execute(
+        "UPDATE dimension SET measure_version_id = page.id FROM page WHERE dimension.page_id = page.guid AND dimension.page_version = page.version"
+    )
+    conn.execute(
+        "UPDATE data_source_in_page SET measure_version_id = page.id FROM page WHERE data_source_in_page.page_guid = page.guid AND data_source_in_page.page_version = page.version"
+    )
+
+    op.alter_column("page", "id", nullable=False)
+    op.alter_column("data_source_in_page", "measure_version_id", nullable=False)
+    op.alter_column("upload", "measure_version_id", nullable=False)
+    op.alter_column("dimension", "measure_version_id", nullable=False)
+
+    # * rename tables:
+    op.rename_table("page", "measure_version")
+    op.rename_table("data_source_in_page", "data_source_in_measure_version")
+
+    # * add constraints back
+    # * Add foreign key and not nullable constraints to `data_source_in_page`, `upload`, `dimension`
+    op.create_primary_key(op.f("measure_version_id_pkey"), "measure_version", ["id", "guid", "version"])
+    op.create_primary_key(
+        op.f("data_source_in_measure_version_data_source_id_pkey"),
+        "data_source_in_measure_version",
+        ["data_source_id", "measure_version_id"],
+    )
+
+    op.create_foreign_key(
+        op.f("measure_version_parent_id_fkey"),
+        "measure_version",
+        "measure_version",
+        ["parent_id", "parent_guid", "parent_version"],
+        ["id", "guid", "version"],
+    )
+    op.create_foreign_key(
```

Comment:
This is really not worth mentioning, but why do some get a line of space between them and others don't?

Reply:
Just me 'logically' (in my head) grouping them into the role they fill, generally. It's more about me making sure I've implemented the right things/number of things in the right order than it anything else.